webpack/assets/javascripts/react_app/redux/API/APIHelpers.js

```diff
@@ -0,0 +1,60 @@
+export const formatSuccess = ({ data, payload, status, error }) => ({
+  status,
+  error,
```

Comment:
formatSuccess always gets `error: null` from what I see, so does it really have to get the `error` param?
(Same for formatPending)

Reply:
We just give the consumer an opportunity to override those attributes,
for example, a "successful" response can return status code 4xx with some data, the consumer than could decide by the response to return an error instead.

Reply:
but when the formatter is called it doesn't give the error from the response so the consumer can't use the error
``` JS
     payload: formatSuccess({
        error: null,
        status: STATUS.RESOLVED,
        data,
        payload,
      })
```

Reply:
yes, but it returns the `data`. lets say that `data` looks like that:
```js
{
  id: 77,
  statusText: "Failed to fetch power status",
}
``` 
this example is returned in status code 2xx, "SUCCESS", but in the UI we might want to display sort of an error by the content of this response.

in this case we give the consumer the ability to override the error:

```js
formatSuccess: ({  error,  status, data, payload  }) => {
 if(data.statusText === xyz){
  return { error: "some-error", status: STATUS.ERROR, ...data }
}
...
```

Reply:
This makes sense, thanks for the explanation.

Reply:
added default values to the formatters helpers for the `status` and `error` in APIHelpers,
so the consumers won't have to pass the error and status but still could override the default values.